jwst/datamodels/model_base.py

```diff
@@ -840,22 +841,41 @@ def get_section(self, name):
 
     @property
     def history(self):
-        return self._instance.setdefault('history', {})
+        """
+        Get the history as a list of entries
+        """
+        # Auto-vivify any part of the hierachy that might be missing
+        tree = self._asdf.tree
+        if 'history' not in tree:
+            tree['history'] = {}
+        if 'entries' not in tree['history']:
+            tree['history']['entries'] = []
+        return tree['history']['entries']
```

Comment:
Perhaps this needs a little bit of discussion. Asdf defines the `history` attribute now as a dict with two keys `extensions` and `entries` where `entries` only represents what  previously was `history`. `DataModel` followed this and included extensions in `DataModel.history` which broke the `append` method as `history` is now a dict. This reverts it back to a list, showing only the `entries`. I think what needs clarification is what is `DataModel.history`? Most people will be probably interested in the entries only. I think if we decide we want to exclude the extensions used this needs to be clarified in the docstrings to avoid going through a similar change of formats again. If we go with `model._asdf.tree['history']['entries'] only, it's probably best to use `AsdfFile.get_history_entries` as this will hide future changes in the implementation.